frontend/packages/dev-console/src/components/add/AddCard.tsx

```diff
@@ -9,14 +12,36 @@ type AddCardProps = {
   title: string;
   items: ResolvedExtension<AddAction>[];
   namespace: string;
+  icon?: CodeRef<React.ReactNode> | string;
 };
 
-const AddCard: React.FC<AddCardProps> = ({ id, title, items, namespace }) => {
+const AddCard: React.FC<AddCardProps> = ({ id, title, items, namespace, icon }) => {
   const isTitleFromItem: boolean = items?.length === 1 && items[0].properties.label === title;
+  const actionIcon = (): JSX.Element => {
+    if (typeof icon === 'string') {
+      return (
+        <img
+          className="odc-add-card__icon odc-add-card__img-icon"
+          src={isValidUrl(icon) ? icon : getImageForIconClass(icon)}
+          aria-hidden="true"
+          alt={title}
+        />
+      );
+    }
+    if (typeof icon !== 'string' && React.isValidElement(icon)) {
+      return (
+        <span className="odc-add-card__icon" aria-hidden="true">
+          {icon}
+        </span>
+      );
+    }
+    return null;
+  };
   return items?.length > 0 ? (
     <Card key={title} className="odc-add-card" data-test={`card ${id}`}>
       {!isTitleFromItem && (
-        <Title size="lg" headingLevel="h2" className="odc-add-card__title" data-test="title">
+        <Title size="lg" headingLevel="h2" className="odc-add-card__title" data-test="title-1">
```

Comment:
Is this `-1` needed? :smirk: 

```suggestion
        <Title size="lg" headingLevel="h2" className="odc-add-card__title" data-test="title">
```